cpp/create_module/algorithm/create.cpp

```diff
@@ -0,0 +1,52 @@
+#include "create.hpp"
+
+void Create::SetElementProp(mgp::Node &element, const mgp::List &prop_key_list, const mgp::List &prop_value_list,
+                            const mgp::RecordFactory &record_factory) {
+  for (size_t i = 0; i < prop_key_list.Size(); i++) {
+    element.SetProperty(std::string(prop_key_list[i].ValueString()), prop_value_list[i]);
+    auto record = record_factory.NewRecord();
+    record.Insert(std::string(kResultProperties).c_str(), std::move(element));
+  }
+}
+
+void Create::ProcessElement(const mgp::Value &element, const mgp::Graph graph, const mgp::List &prop_key_list,
+                            const mgp::List &prop_value_list, const mgp::RecordFactory &record_factory) {
+  if (element.IsNode()) {
+    auto node = element.ValueNode();
+    SetElementProp(node, prop_key_list, prop_value_list, record_factory);
+    return;
+  }
+  if (element.IsInt()) {
+    auto node = graph.GetNodeById(mgp::Id::FromInt(element.ValueInt()));
```

Comment:
This may throw. You should wrap this ProcessElement in try catch block and in case of error, set Error message in case of error. Check Record::SetErrorMessage method

Reply:
Isn't try-catch in main function(SetProperties) enough since that is the function which is calling ProcessElement? Right now if I send unexisting node id I get "Query failed: create.set_properties: Node with ID 90 not found!" error that seems informative enough? Or should I put something else?

Reply:
You are right, didn't see that, sorry